### PR TITLE
beacon/light: blsync'er alternate

### PR DESCRIPTION
**Pull request by lightclient**
*Parts of this is copied directly from Zsolt's blsync implementation.*

This is mostly working now. If you want to try it out you can run the following:

```console
$ geth --sepolia --beacon.api=https://lodestar-sepolia.chainsafe.io --beacon.wsc=$(curl -X GET 'https://lodestar-sepolia.chainsafe.io/eth/v1/beacon/headers/finalized' | jq ".data.root" | sed 's/"//g') --log.vmodule="beacon/light/*=5"
```

Only supports Sepolia at the moment, but easy to add some more configs if this is a direction we want to go.

I think this conveys my perspective that we can drive geth via a light client w/o adding much complexity. This change is only around 1600 diff to the LOC in geth whereas the other approach is 5000+ LOC.

Obviously there are limitations of this approach and is nowhere as fully featured as Zsolt's- but as I've argued in the past, I don't think we should provide a thorough light client picture for users and so this PR provides all the functionality that we could need in an extremely simple and easy to understand format.

### CODE REVIEW

File: beacon/light/chain.go [hunk 1]
```diff
@@ -159,3 +164,114 @@ func (c *LightClient) getExecutableData(head common.Hash) (*engine.ExecutableDat
...
+			StateRoot:     common.Hash(p.StateRoot),
+			ReceiptsRoot:  common.Hash(p.ReceiptsRoot),
+			LogsBloom:     p.LogsBloom[:],
+			Random:        p.PrevRandao,
+			Number:        p.BlockNumber,
+			GasLimit:      p.GasLimit,
+			GasUsed:       p.GasUsed,
+			Timestamp:     p.Timestamp,
+			ExtraData:     p.ExtraData,
+			BaseFeePerGas: nil, // TODO: convert this []uint64 correctly to big.Int
```
Comment (MariusVanDerWijden) [inline beacon/light/chain.go:247]: I think you correctly convert this in the other cases

Comment (MariusVanDerWijden) [reply beacon/light/chain.go:247]: nvm, just saw that this is just moved

File: beacon/beaclient/client.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		return nil, fmt.Errorf("failed to read response: %w", err)
+	}
+	var u types.LightClientUpdate
+	if err := json.Unmarshal(b, &u); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal data: %v", err)
+	}
+	return &u, nil
+}
+
+func (c *Client) GetFinalityUpdate() (*types.LightClientUpdate, error) {
```
Comment (MariusVanDerWijden) [inline beacon/beaclient/client.go:114]: `GetRangeUpdate` `GetOptimisticUpdate` and `GetFinalityUpdate` kinda do the same thing, only for different URLs. We could factor that into a private method to reduce the amount of copy-pasta

Comment (lightclient) [reply beacon/beaclient/client.go:100]: fixed!

File: beacon/light/chain.go (hunk 1 above)
Comment (lightclient) [reply beacon/light/chain.go:256]: i'm not exactly sure when I put this comment here, but it seems like the deneb rendition from the eth2 lib has `BaseFeePerGas` as a `uint256` so I can just convert that to `big.Int`.

**Comment by lightclient** [comment]
closing favor of #28822

**Pull request closed without merging**